examples/wallet_with_pincode/wallet_with_pincode.v.py

```diff
@@ -0,0 +1,77 @@
+# Simple wallet with PIN protection
+
+# bad guys can not get money out of this contract-wallet
+# even if your secret key has been compromised
+# if they do not know the `secret`
+# (pre-image of the hash that is stored in contract).
+
+# Owners wallet.
+owner: public(address)
+
+# Hash of a secret number that allows user
+# to operate with funds from this contract.
+secret_hash: public(bytes32)
+
+# Assign owner to contract creator.
+def __init__():
+    self.owner = msg.sender
+
+# This function allows to reset secret
+# each secret should be used only once
+# since everyone can track your transaction data
+# in blockchain.
+def set_secret(_secret: num, _new_secret: bytes32):
+    # Only owner can execute this.
+    assert msg.sender == self.owner
+
+    # Prove ownership with a secret phrase.
+    assert sha3(as_bytes32(_secret)) == self.secret_hash
+
+    # Assign a new hash of the secret
+    # pre-image of the hash is not known for third parties
+    # since it was not stored in blockchain
+    # until the first successful attempt to unlock this contract.
+    self.secret_hash = _new_secret
+
+# This function allows to deposit funds
+# into this contract.
+# `@payable` decorator allows function to receive Ether
+@payable
+def deposit():
+    return
+
+# A function that allows anyone
+# to check which hash matches the entered secret.
+# `@constant` decorator allows call this function
+# without broadcasting a transaction to the blockchain.
+@constant
+def secret_from_num(_secret: num) -> bytes32:
```

Comment:
Name is a little confusing.

Can this be restructured into `check_secret()` that performs the same function `set_secret()`, `withdraw()`, and `call()` check for in their execution, that way 1. you're not repeating yourself and 2. This operation is clearer to it's intent?